### PR TITLE
chore: Add an end to end test workflow

### PR DESCRIPTION
**Pull request by ItayLevyOfficial**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [x]  Targeted PR against correct branch
- [x]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x]  Linked to Github issue with discussion and accepted design
- [x]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Execute IBC relay tester
+        run: |
+          chmod +x ./scripts/ibc_relay_tester.sh
+          DEST_ADDR=$HUB_SEQ_ADDR ROLLAPP_ID=$ROLLAPP_ID ROLLER_CONFIG_PATH=$ROLLER_CONFIG_PATH ./scripts/ibc_relay_tester.sh
+
+      - name: Terminate the rollapp
+        if: always()
+        run: |
+          kill $ROLLAPP_PID
+      - name: Terminate the hub
```
Comment (mtsitrin) [inline .github/workflows/e2e.yml:72]: what about terminating the `ibc_relay_tester` script?

File: scripts/ibc_relay_tester.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+if [ -z "$DEST_ADDR" ] || [ -z "$ROLLAPP_ID" ] || [ -z "$ROLLER_CONFIG_PATH" ]; then
+    echo "Error: One or more required environment variables (DEST_ADDR, ROLLAPP_ID, ROLLER_CONFIG_PATH) are unset."
+    exit 1
+fi
+
+TIMEOUT=1800 # 30 minutes
+
+check_balance() {
+    BALANCE=$(/usr/local/bin/dymd query bank balances $DEST_ADDR | grep "ibc/")
```
Comment (mtsitrin) [inline scripts/ibc_relay_tester.sh:15]: what about --home for the dymd?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (ItayLevyOfficial) [reply .github/workflows/e2e.yml:72]: it quits alone after a successful transaction / timeout

File: scripts/ibc_relay_tester.sh (hunk 2 above)
Comment (ItayLevyOfficial) [reply scripts/ibc_relay_tester.sh:15]: You don't need it for querying the balance

Comment (mtsitrin) [reply scripts/ibc_relay_tester.sh:15]: so u need --node?

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**